### PR TITLE
chore: implement workspace dependencies

### PR DESCRIPTION
**Pull request by rjzak**
Closes https://github.com/profianinc/steward/issues/121
Depends on https://github.com/profianinc/steward/pull/146

Signed-off-by: Richard Zak <richard@profian.com>

### CODE REVIEW

**Comment by rvolosatovs** [comment]
@rjzak please rebase

**Review by rvolosatovs — CHANGES_REQUESTED** [review]
Please move `features =` blocks into the actual crate dependencies that need them

File: Cargo.toml [hunk 1]
```diff
@@ -7,21 +7,63 @@ version = "0.2.0"
...
+der = { version = "0.6", features = ["std"], default-features = false }
+flagset = { version = "0.4.3", default-features = false }
+hex = { version = "0.4.3", features = ["alloc"], default-features = false }
+http = { version = "^0.2.6", default-features = false }
+hyper = { git = "https://github.com/rjzak/hyper", branch = "wasi_wip", features = ["http1", "server"], default-features = false }
+memoffset = { version = "0.7.1", default-features = false }
+p256 = { version = "0.11", features = ["ecdsa", "std", "pem"], default-features = false }
+p384 = { version = "0.11", features = ["ecdsa", "std", "pem"], default-features = false }
+rand = { version = "0.8", features = ["std"], default-features = false }
+rsa = {version = "0.7.2", features = ["std"], default-features = false }
```
[the remark this answers is not in the document]
Comment (rjzak) [reply Cargo.toml:30]: How about have the features here when all places that use this crate use these features?

[the remark this answers is not in the document]
Comment (rvolosatovs) [reply Cargo.toml:30]: That's true today, but if that was ever to change is would be very difficult to track

[the remark this answers is not in the document]
Comment (rjzak) [reply Cargo.toml]: Moved.

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**